MantleTests/MTLModelDifferenceSpec.m

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) 2020 Lightricks. All rights reserved.
+// Created by Maxim Grabarnik.
+
+#import <Mantle/Mantle.h>
+#import <Nimble/Nimble.h>
+#import <Quick/Quick.h>
+
+#import "MTLModel+Difference.h"
+#import "MTLTestModel.h"
+
+QuickSpecBegin(MTLModelDifferenceSpec)
+
+it(@"should raise on mismatching argument type", ^{
+	MTLTestModel *model = [[MTLTestModel alloc] init];
+	expect(model).notTo(beNil());
+	expect([model differenceFrom:@1]).to(raiseException());
+
+});
+
+it(@"should compute different fields", ^{
+	NSDictionary *values1 = @{
+		@"name": @"foo",
+		@"count": @(4),
+	};
+
+	NSError *error = nil;
+	MTLTestModel *model1 = [[MTLTestModel alloc] initWithDictionary:values1 error:&error];
+	expect(model1).notTo(beNil());
+
+	NSDictionary *values2 = @{
+		@"count": @(5),
+		@"nestedName": @"moo",
+	};
+
+	MTLTestModel *model2 = [[MTLTestModel alloc] initWithDictionary:values2 error:&error];
+	NSArray *difference = [model1 differenceFrom:model2];
+	expect(difference).to(equal(@[
+    @"count:",
+    @"|\tExpected: 4",
+    @"|\tActual\t: 5",
+		@"name:",
+		@"|\tExpected: foo",
+		@"|\tActual\t: (null)",
+		@"nestedName:",
+		@"|\tExpected: (null)",
+		@"|\tActual\t: moo"
+	]));
+});
+
+it(@"should compute different fields of sub-object", ^{
+	NSDictionary *values1 = @{
+		@"name": @"foo",
+	};
+	NSDictionary *values2 = @{
+		@"name": @"moo",
+	};
+
+	NSError *error = nil;
+	MTLTestModel *submodel1 = [[MTLTestModel alloc] initWithDictionary:values1 error:&error];
+	MTLTestModel *submodel2 = [[MTLTestModel alloc] initWithDictionary:values2 error:&error];
+
+	MTLTestModel *model1 = [[MTLTestModel alloc] initWithDictionary:@{ @"weakModel": submodel1 }
+																														error:&error];
+
+	MTLTestModel *model2 = [[MTLTestModel alloc] initWithDictionary:@{ @"weakModel": submodel2 }
+																														error:&error];
+
+	NSArray *difference = [model1 differenceFrom:model2];
+	expect(difference).to(equal(@[
+		@"weakModel:",
+		@"|\tname:",
+		@"|\t|\tExpected: foo",
+		@"|\t|\tActual\t: moo"
+	]));
```

Comment:
indent should be with 2 spaces, not 4 same as [here](https://github.com/Lightricks/Mantle/blob/develop/MantleTests/MTLJSONAdapterSpec.m#L25-L28)